packages/components/src/ConditionalFilter/GroupFilter.tsx

```diff
@@ -19,26 +18,62 @@ import { isChecked, calculateSelected, getGroupMenuItems, getMenuItems, convertT
 import groupType from './groupType';
 import './group-filter.scss';
 
-const Group = ({
+interface GroupFilterItem {
+  id?: string;
+  value?: string;
+  type?: 'treeView' | 'checkbox' | 'radio' | 'button' | 'plain';
+  label?: React.ReactNode;
+}
+
+interface GroupFilterGroupItem {
+  id?: string;
+  tagKey?: string;
+}
+
+export interface GroupFilterGroup {
+  groupSelectable?: boolean;
+  type?: 'treeView' | 'checkbox' | 'radio' | 'button' | 'plain';
+  value?: string;
+  label?: React.ReactNode;
+  items?: Array<GroupFilterGroupItem>;
+}
+
+export interface GroupProps {
+  placeholder?: string;
+  onShowMore?: (event: React.MouseEvent | React.KeyboardEvent | MouseEvent) => void;
+  showMoreTitle?: React.ReactNode;
+  showMoreOptions?: Record<string, unknown>;
+  items?: Array<GroupFilterItem>;
+  className?: string;
+  selected?: Record<string, Record<string, Record<string, unknown>>>;
+  isFilterable?: boolean;
+  filterBy?: string;
+  onFilter?: () => void;
+  groups?: Array<GroupFilterGroup>;
+  onChange: () => void;
+  selectedTags?: Record<string, unknown>;
+}
+
+const Group: React.FunctionComponent<GroupProps> = ({
   placeholder,
   onShowMore,
   showMoreTitle,
   showMoreOptions,
   items,
-  filterBy,
+  filterBy = '',
   onFilter,
   className,
   groups = [],
   onChange,
-  selected,
+  selected = {},
   isFilterable,
 }) => {
-  const [stateSelected, setStateSelected] = useState({});
-  const [searchString, setSearchString] = useState('');
-  const [isOpen, setIsOpen] = useState(false);
-  const toggleRef = useRef();
-  const containerRef = useRef();
-  const menuRef = useRef();
+  const [stateSelected, setStateSelected] = useState<Record<string, unknown>>({});
+  const [searchString, setSearchString] = useState<string>('');
+  const [isOpen, setIsOpen] = useState<boolean>(false);
```

Comment:
You don't have to explicitly set type for primitive types. TS will infer the type based on the default value.
```diff
- const [isOpen, setIsOpen] = useState<boolean>(false);
+ const [isOpen, setIsOpen] = useState(false);
```